Refactoring of both readme and project code (#2)
ravendb/ravendb-jvm-tutorials: README.md

public final class RavenDBDocumentStore {
    private static IDocumentStore store;

    static {

        //Create new Document Store with the url of the RavenDB Server
        //and with `Hospital` set as the default database
        store = new DocumentStore(new String[]{"http://127.0.0.1:8080"},
                                  "Hospital");

        //Edit the conventions to enable optimistic concurrency
        DocumentConventions conventions = store.getConventions();
        conventions.setUseOptimisticConcurrency(true);

        store.initialize();
    }

    //Return the single Document Store instance 
    public static IDocumentStore getStore() {
        return store;
    }

}
```
## Session and Unit of Work Pattern  

Any operation we want to perform starts by obtaining a new _Session_ object from the Document Store.
The Session implements the [Unit of Work](https://martinfowler.com/eaaCatalog/unitOfWork.html)
pattern. This has several implications:  
* The Session batches requests to the server to minimize the number of round trips over the network.  
* The Session tracks changes for all the entities that it has either loaded from or stored to the database, and commits 

> them as an ACID transaction when `[session].SaveChanges()` is called.  

Reviewer: ```suggestion
them as an ACID transaction when `[session].saveChanges()` is called.  
```

total amount of results to a query (to calculate the total number of pages) and then to lazily load the results in batches 
of 10 as the user navigates from page to page. We can use `statistics()` to access useful data about the query, including 
the total number of results. For the patients grid, the corresponding attachments are also obtained and streamed into a 
byte array.

![Patient Paging](/screenshots/p_paging.png)

`getPatientsList()`, with error handling omitted for brevity:
```java
public Pair<Collection<PatientWithPicture>, Integer> getPatientsList(int offset, int limit, boolean order) {
    Reference<QueryStatistics> statsRef = new Reference<>();
    IDocumentQuery<Patient> query = session.query(Patient.class)
                                           .skip(offset)
                                           .take(limit)
                                           .statistics(statsRef);

    if (order) {
        query.orderBy("birthDate");
    }

    Collection<Patient> list = query.toList();
    int totalResults = statsRef.value.getTotalResults();

    Collection<PatientWithPicture> patientWithPictures = new ArrayList<>();
}
```

## BLOB Handling - Attachments

> Binary data that cannot be stored as JSON (such as images, audio, etc.) can be associated with a document as one or more _attachments_.

Reviewer: Maybe not "cannot be stored", but should be stored separately

Author: ```suggestion
Binary data that needs to be stored in addition to the JSON content (such as images, audio, etc.) can be associated with a document as one or more _attachments_.
```

An example are the profile pictures of Patients. This POJO represents attachments on the client side:
```java
public class Attachment {

    String name;

>     //String mimeType;

Reviewer: ```suggestion
```

        profilePicture.setBytes(bytes);
        profilePicture.setName(ATTACHMENT_NAME);
        
        return profilePicture;
    } catch (IOException e) {
        logger.log(Level.SEVERE, "", e);
    }

    return null;
}
```

## Queries
In RavenDB, a query can only be satisfied by an index. You can create your own indexes, but if no appropriate index exists, 
RavenDB will automatically create one on the fly. Indexes learn from each new query, and are continuously optimized to 
satisfy all previous queries made to a given collection. In this way RavenDB is built to optimize the speed of reads with minimal 
administration. (By default, indexes are updated asynchronously so that they don't impact write speed. In most use cases 
this improves performance, but you can always have write operations wait for the index to update before returning using 
`session.advanced().waitForIndexesAfterSaveChanges();`.)  

Queries are created by chaining methods on `session.query()`, and executed by calling `.toList()`. Before being sent to the server, every
query is translated to [RQL (RavenDB Query Language)](https://ravendb.net/docs/article-page/4.2/java/indexes/querying/what-is-rql) - 

> our SQL-like language. RQL was designed to expose RavenDB's query pipeline to the user. You can write your queries in RQL

Reviewer: ```suggestion
our SQL-like language, RQL was designed to expose RavenDB's query pipeline to the user. You can write your queries in RQL
```

Author: It's hard to read in markdown because of the link in the middle, but "our SQL-like language" belongs to the previous sentence, not the next one:
> ...every query is translated to [RQL (RavenDB Query Language)]() - our SQL-like language. RQL was designed...
